### PR TITLE
[Tests] Add missing tests for NWProtocolStack and fix NWParameters tests.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Modifying the stack of the connection does make the tests for NWParameters to fail, since they'll not wait for the connection to be done. I had to modify the NWParamtersTests to isolate them from the changes done in the stack in other fixtures.

### CODE REVIEW

File: tests/monotouch-test/Network/NWParametersTest.cs [hunk 1]
```diff
@@ -29,52 +29,74 @@ public class NWParametersTest {
...
 
-		void ConnectionStateHandler (NWConnectionState state, NWError error)
-		{
-			var errno = (SslStatus)(error != null ? error.ErrorCode : 0);
-			switch (state){
-			case NWConnectionState.Ready:
-				connectedEvent.Set ();
+ 		void ConnectionStateHandler (NWConnectionState state, NWError error)
+ 		{
+ 			Console.WriteLine ($"State is {state} and error {error}");
```
Comment (spouliot) [inline tests/monotouch-test/Network/NWParametersTest.cs]: remove, or comment, debugging output

File: tests/monotouch-test/Network/NWProtocolStackTest.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+		[SetUp]
+		public void SetUp ()
+		{
+			options = new List<NWProtocolOptions> (); 
+		}
+
+		void ConnectionStateHandler (NWConnectionState state, NWError error)
+		{
+			Console.WriteLine ($"State is {state} and error {error}");
```
Comment (spouliot) [inline tests/monotouch-test/Network/NWProtocolStackTest.cs]: same

File: tests/monotouch-test/Network/NWProtocolStackTest.cs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+		[TestFixtureSetUp]
+		public void Init ()
+		{
+			TestRuntime.AssertXcodeVersion (10, 0);
+			// we want to use a single connection, since it is expensive
+			connectedEvent = new AutoResetEvent(false);
+			host = "www.google.com";
+			using (var parameters = NWParameters.CreateUdp ())
+			using (var endpoint = NWEndpoint.Create (host, "80"))
+			{
```
Comment (spouliot) [inline tests/monotouch-test/Network/NWProtocolStackTest.cs]: minor: `{` on same line as `using`

File: tests/monotouch-test/Network/NWProtocolStackTest.cs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+			using (var o = stack.InternetProtocol)
+			{
+				o.IPSetUseMinimumMtu (true); // should not crash
+			}
+		}
+
+	}
+ }
+
+ #endif
```
Comment (rolfbjarne) [inline tests/monotouch-test/Network/NWProtocolStackTest.cs]: Minor: missing EOL for last line.

**Pull request merged**